scrapbook/api.py

```diff
@@ -84,30 +85,48 @@ def glue(name, scrap, encoder=None, display=None):
 
     # Only store data that can be stored (purely display scraps can skip)
     if encoder != "display":
-        data = {
-            GLUE_PAYLOAD_FMT.format(encoder=encoder): scrap_to_payload(
-                encoder_registry.encode(Scrap(name, scrap, encoder))
-            )
-        }
-        metadata = {"scrapbook": dict(name=name)}
-        # IPython.display.display takes a tuple of objects as first parameter
-        # `http://ipython.readthedocs.io/en/stable/api/generated/IPython.display.html#IPython.display.display`
+        data, metadata = _data_scrap_display(name, scrap, encoder)
         ip_display(data, metadata=metadata, raw=True)
 
     # Only display data that is marked for display
     if display:
-        display_kwargs = {}
-        if isinstance(display, (list, tuple)):
-            display_kwargs = {"include": display}
-        elif isinstance(display, dict):
-            display_kwargs = display
-        data, metadata = IPython.core.formatters.format_display_data(
-            scrap, **display_kwargs
-        )
-        metadata["scrapbook"] = dict(name=name)
+        data, metadata = _display_scrap_display(name, scrap, display)
```

Comment:
Not sure about this naming `_data_scrap_display` and `_display_scrap_display`? They feel a bit awkward and not quite sure what they do. Perhaps `_prepare_scrap_display` (prepare for data) and `_draw_up_scrap_display` or `_assemble_scrap_display` (draw up or assemble for display).

Reply:
Agreed. I backed off on some of the refactor here and just made the single shared piece accessible. See if the new name makes more sense?